client/dashboard/app/queries/domain-dns-records.ts

```diff
@@ -42,3 +43,21 @@ export const domainDnsEmailMutation = ( domainName: string ) =>
 			} );
 		},
 	} );
+
+export const domainDnsApplyTemplateMutation = ( domainName: string ) =>
+	mutationOptions( {
+		mutationFn: ( params: {
```

Comment:
[nit] I think we don't need to explicitly use named parameter `param` here if it's never being used (we can just spread everything).